cpp/include/Ice/ProxyFunctions.h

```diff
@@ -15,19 +15,21 @@ namespace Ice
     /**
      * Verifies that a proxy received from the client is not null, and throws a MarshalException if it is.
      * @param prx The proxy to check.
+     * @param file The source file name.
+     * @param line The source line number.
      * @param current The Current object for the invocation.
      * @throw MarshalException If the proxy is null.
      * */
     template<typename Prx, std::enable_if_t<std::is_base_of<ObjectPrx, Prx>::value, bool> = true>
-    void checkNotNull(std::optional<Prx> prx, const Current& current)
+    void checkNotNull(std::optional<Prx> prx, const char* file, int line, const Current& current)
```

Comment:
In case you are wondering, defaulting file and line to = __FILE__ and __LINE__ doesn't work - it gives you the file and line in ProxyFunctions.h.